### DISCUSSION TITLE
human_idx setting

### DISCUSSION THREAD

**Discussion by leijue222** [fredzzhang/spatio-attentive-graphs#82] [Q&A] [answered]
Hi, @fredzzhang 
I don't understand the setting of the parameters and would like to ask for your help.

For VCOCO datasets:
In UPT, you set human_idx=0
https://github.com/fredzzhang/upt/blob/192743045e4c5f6827299fdc15809f5b97ac1b8e/main.py#L63-L70

In SCG, you set human_idx=1
https://github.com/fredzzhang/spatially-conditioned-graphs/blob/a44093ea0d77e0d4a27b1694c18df2fec9af5770/main.py#L73-L75

I wonder what causes setting human_idx to be different. 
I understand that it is set to 0, because the COCO dataset classifies the human class as 0, but why is human_idx set to 1 in SCG?

**Comment by fredzzhang** [comment, marked as answer]
Hi @leijue222,

It's just a convention inherited from the object detector. SCG used Faster R-CNN, which has human class encoded as `1` with background class as `0`. For UPT, detector has been changed to DETR, which encodes human class as `0` and background class as `N`.

Fred.
